File: remove-duplicates-from-sorted-array-2/python/Solution.py
from typing import List


class Solution:
  def removeDuplicates(self, nums: List[int]) -> int:
    #         Input: nums = [0,0,1,1,1,2,2,3,3,4]
    # Output: 5, nums = [0,1,2,3,4,_,_,_,_,_]

    originalPosition = 0
    for i in range (1, len(nums)):
      current = nums[i]
      if (current != nums[originalPosition]):
        originalPosition+=1
        nums[originalPosition] = current


    return originalPosition + 1
  
  #A follow up might be what if you're allowed at most 2 of the same elements 
  def removeDuplicatesTweak(self, nums: List[int]) -> int:
    #         Input: nums = [0,1,1,1,2,2,3,3,3,3,4,4]
    # Output: 9, nums = [0,1,2,3,4,_,_,_,_,_]

    originalPosition = 0
    for i in range (1, len(nums)):
      current = nums[i]
      if (originalPosition == 0 or nums[originalPosition - 1] != current):
        originalPosition+=1
        nums[originalPosition] = current


    return originalPosition + 1

File: remove-duplicates-from-sorted-array-2/cpp/Solution.cpp
#include "Solution.h"
#include <vector>

int Solution::removeDuplicates(std::vector<int>& nums) {
//   Input: nums = [0,0,1,1,1,1,2,3,3]
// Output: 7, nums = [0,0,1,1,2,3,3,_,_]

  int originalPosition = 0;
  for (int i = 1; i < nums.size(); i++) {
    int current = nums[i];
    if (originalPosition == 0 || current != nums[originalPosition]) {
      originalPosition++;
      nums[originalPosition] = current;
    }
  }
  return originalPosition + 1;

}



